lib/editor/components/ExceptionDateRange.js

```diff
@@ -0,0 +1,191 @@
+// @flow
+
+import React, {Component} from 'react'
+import {DropdownButton, FormGroup, MenuItem} from 'react-bootstrap'
+import DateTimeField from 'react-bootstrap-datetimepicker'
+import moment from 'moment'
+
+import {updateActiveGtfsEntity} from '../actions/active'
+import { getComponentMessages } from '../../common/util/config'
+import {modifyRangeOfDates, updateDates} from '../../common/util/exceptions'
+import type {ExceptionDate, ScheduleException, ScheduleExceptionDateRange} from '../../types'
+import type {EditorValidationIssue} from '../util/validation'
+
+import {inputStyleProps} from './ExceptionDate'
+import ExceptionValidationErrorsList from './ExceptionValidationErrorsList'
+
+type Props = {
+  activeComponent: string,
+  activeEntity: ScheduleException,
+  index: number,
+  range: ScheduleExceptionDateRange,
+  updateActiveGtfsEntity: typeof updateActiveGtfsEntity,
+  validationErrors: Array<EditorValidationIssue>,
+  validationState: string
+}
+
+type State = {
+  forceErrorState: boolean
+}
+
+export const sortDates = (dates: Array<ExceptionDate>): void => { dates.sort((a, b) => moment(a).diff(moment(b))) }
+
+export default class ExceptionDateRange extends Component<Props, State> {
+  messages = getComponentMessages('ExceptionDateRange')
+
+  state = {
+    forceErrorState: false
+  }
+
+  // This is kept as an arrow function in order to access the component's `this`
+  _onRangeUpdate = (millis: number | string, isStartDate: boolean) => {
+    if (millis === 'Invalid date') return // Invalid date thrown from Datetimepicker is usually caused by typing
+    const {activeComponent, activeEntity, range, updateActiveGtfsEntity} = this.props
+    const {endDate, startDate} = range
+
+    const newRangeBoundary = moment(+millis)
+    const startMoment = moment(startDate)
+    const endMoment = moment(endDate)
+
+    // This prevents the user from inputting invalid data. Validation is not required since this behaviour
+    // is fully prevented (see also use of minDate and maxDate in DateTimePickers)
+    if (
+      (isStartDate && (isStartDate && newRangeBoundary.diff(endMoment) > 0)) ||
```

Comment:
`isStartDate`is checked twice

Reply:
Good catch! => https://github.com/ibi-group/datatools-ui/pull/1032/commits/fa615c7b2841cfec63168749834851850ac66083